solvers.go

```diff
@@ -648,6 +651,18 @@ type Challenge struct {
 	data interface{}
 }
 
+// challengeKey determines the key for a challenge Identifier
+func challengeKey(chal acme.Challenge) string {
+	key := chal.Identifier.Value
+	if chal.Type == acme.ChallengeTypeTLSALPN01 && chal.Identifier.Type == "ip" {
+		k, err := dns.ReverseAddr(chal.Identifier.Value)
+		if err == nil {
+			key = k[:len(k)-1] // strip off .
+		}
+	}
+	return key
```

Comment:
This might just be me nit-picking, but I feel like this flows a little better. :smile: 

```suggestion
	if chal.Type == acme.ChallengeTypeTLSALPN01 && chal.Identifier.Type == "ip" {
		reversed, err := dns.ReverseAddr(chal.Identifier.Value)
		if err == nil {
			return reversed[:len(reversed)-1] // strip off '.'
		}
	}
	return chal.Identifier.Value
```

Reply:
👍